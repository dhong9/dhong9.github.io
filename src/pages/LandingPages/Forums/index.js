import { useEffect, useState } from "react";

// Sections components
import BaseLayout from "layouts/sections/components/BaseLayout";
import ForumHeader from "components/ForumHeader";

// Services
import { getCategories } from "services/categoriesService";
import { getPosts } from "services/postsService";
import MainFeaturedPost from "components/MainFeaturedPost";

function Forums() {
  const [categories, setCategories] = useState([]);
  const [mainFeaturedPosts, setMainFeaturedPosts] = useState([]);

  useEffect(() => {
    // Get categories
    getCategories(({ data: { results } }) => {
      setCategories(results);
    });

    // Get posts
    getPosts(({ data: { results } }) => {
      const mainFeaturedList = [];
      for (const { postName, body, mainFeatured } of results) {
        // A post cannot be both main featured and just featured
        if (mainFeatured) {
          const post = {
            postName,
            body,
            image: "https://source.unsplash.com/random",
            imageText: "main image description",
            linkText: "Continue reading...",
          };
          mainFeaturedList.push(post);
        }
      }
      setMainFeaturedPosts(mainFeaturedList);
    });
  }, []);

  return (
    <BaseLayout
      title="Doughnut Rider"
      breadcrumb={[{ label: "Doughnut Rider", route: "/doughnutRider" }]}
    >
      <ForumHeader categories={categories} />
      {mainFeaturedPosts.map((mainFeaturedPost) => (
        <MainFeaturedPost key={mainFeaturedPost.id} post={mainFeaturedPost} />
      ))}
    </BaseLayout>
  );
}

export default Forums;
