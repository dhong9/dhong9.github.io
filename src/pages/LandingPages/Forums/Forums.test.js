// Unit test libraries
import React from "react";
import renderer from "react-test-renderer";

// Component to test
import Forums from "pages/LandingPages/Forums";

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
  jest.clearAllMocks();
});

// Mocks
jest.mock("react-monaco-editor", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Editor</div>),
  };
});
jest.mock("components/MKBox", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>Mock Box</div>),
  };
});
jest.mock("draft-convert", () => {
  return {
    convertFromHTML: jest.fn(),
    convertToRaw: jest.fn(),
  };
});

describe("Forums", () => {
  it("renders", () => {
    useContextMock.mockReturnValue("Test Value");

    const component = renderer.create(<Forums />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
