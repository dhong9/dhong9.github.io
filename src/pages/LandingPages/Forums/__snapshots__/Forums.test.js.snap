// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forums renders 1`] = `
<div>
  Mock Box
</div>
`;
